sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/data/PointData.java

```diff
@@ -38,4 +39,8 @@
    * @return the attributes associated with this {@code Point}.
    */
   Attributes getAttributes();
+  /**
+   * (Optional) List of exemplars collected from measurements that were used to form the data point.
+   */
+  Collection<Exemplar> getExemplars();
```

Comment:
We use `List` for similar types in `SpanData` let's go with it here, there doesn't seem to be much gain in the more generic type

Reply:
I was planning to use a highly concurrent-friendly collection type that is NOT a `List` here in the implementation.   If it's all the same, I don't see a big gain for this being `List`, but I see a big downside in choice of collection for sampling.

Reply:
Can you describe the structure more to understand better? And is it that hard to present a List view of it on export (when presumably everything is mostly frozen?). Consistency within the SDK is better unless there is a big downside

Reply:
So two things:

1. I actually want this to be `Iterable` (I forked this PR earlier than the prototype), which I'll update if you agree.
2. I'm using a linked-list that allows prepend-only compare-and-swap behavior for lock-less addition.   This *is* a list where order is reversed from expectations *and* the `get(index)` method would be highly inefficient to use.   The primary reason to use this list is to have a highly-concurrent append directly from the sampler and it's immutable so easy to share/export to OTLP.

You can see an earlier prototype here: https://github.com/jsuereth/opentelemetry-java/blob/wip-metrics-api/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/aggregator/ExemplarList.java

I can implement the List interface (with lots of exception throwing) but I feel it's pretty disingenuous to give people a "get" method that's highly inefficient.  Probably too much "Scala" collection design in me.   PTAL and let me know your thoughts here.

Reply:
Yeah I wouldn't worry too much about deceptive get methods which have already been presented to us by Java in `LinkedList`.

But given metrics is a concurrent thing, while spans we expect generally not to be, I can see where using different collections can come in handy so a more generic type seems ok. Given the choice between `Collection` and `Iterable` I would pick `Collection` - Java doesn't even give easy conversion to `Stream` for `Iterable` so it's quite an anemic return type for a public API.

By the way, I don't know if you already looked at it, but if it's possible to use, we should add to our shading-in of JCTools a structure instead of reimplementing, this one seems similar to what you describe.

https://github.com/JCTools/JCTools/blob/master/jctools-core/src/main/java/org/jctools/queues/MpscLinkedQueue.java

Currently we shade into the trace artifact but no reason not to move that to common

https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk/trace-shaded-deps

Reply:
Alright I'll go with List<> then for consistency and just document the hell out of concurrency/efficiency going forward.

Thanks for the link to the MPscLinkedQueue!  We don't really want a queue, and we may wind up with a "keep-latest-but-have-slots" array-of-atomic-references going forward.  Once the exemplar sampling is finalized, we'll have a beter idea what we need (see: https://github.com/open-telemetry/opentelemetry-specification/issues/1797#issuecomment-876431458)